### PR TITLE
Dns Update (RFC2136) catalog entry

### PR DESCRIPTION
**Pull request by digitalLumberjack**
Add the rfc2136dns provider.

Related and depends on https://github.com/rancher/external-dns/pull/35
I set the image to rancher/external-dns:v0.4.2, thinking you will release a new version if you merge the external-dns PR.

### CODE REVIEW

**Comment by cloudnautique** [comment]
@alena1108, have we done a recent release of the external-dns?

**Comment by alena1108** [comment]
@cloudnautique no, we haven't. There are several PRs still in review. So all external-dns related PRs should be marked as WIP until we release the new image.

File: templates/dnsupdate-rfc2136/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  expose:
+   - 1000
+  command:
+  - -provider
+  - RFC2136
+  environment:
+    DNS_HOST: ${DNS_HOST}
+    DNS_PORT: ${DNS_PORT}
+    TSIG_KEYNAME: ${TSIG_KEYNAME}
+    TSIG_KEY: ${TSIG_KEY}
```
Comment (janeczku) [inline templates/dnsupdate-rfc2136/0/docker-compose.yml]: @digitalLumberjack Can you please update the environment variables to match those in the rfc2136 provider?
`DNS_HOST` => `RFC2136_HOST`
`DNS_PORT` => `RFC2136_PORT`
`TSIG_KEYNAME` => `RFC2136_TSIG_KEYNAME`
`TSIG_KEY` => `RFC2136_TSIG_SECRET`

See https://github.com/rancher/external-dns/blob/master/providers/rfc2136/rfc2136.go#L26-L41

File: templates/dnsupdate-rfc2136/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+rfc2136dns:
+  image: rancher/external-dns:v0.4.2
+  expose:
+   - 1000
+  command:
+  - -provider
+  - RFC2136
```
Comment (janeczku) [inline templates/dnsupdate-rfc2136/0/docker-compose.yml]: Name slug should be lowercase: `rfc2136`
See https://github.com/rancher/external-dns/blob/master/providers/rfc2136/rfc2136.go#L23

**Comment by digitalLumberjack** [comment]
Done :)

File: templates/dnsupdate-rfc2136/0/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+rfc2136dns:
+  image: rancher/external-dns:v0.4.2
```
Comment (janeczku) [inline templates/dnsupdate-rfc2136/0/docker-compose.yml]: rancher/external-dns:v0.5.0

File: templates/dnsupdate-rfc2136/config.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+name: DNS Update (RFC2136)
+description: |
+  Rancher External DNS service powered by any RFC2136 compatible DNS
+version: v0.4.2-rancher1
```
Comment (janeczku) [inline templates/dnsupdate-rfc2136/config.yml]: v0.5.0-rancher1

File: templates/dnsupdate-rfc2136/0/rancher-compose.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+.catalog:
+  name: "DNS Update (RFC2136)"
+  version: "v0.4.2-rancher1"
```
Comment (janeczku) [inline templates/dnsupdate-rfc2136/0/rancher-compose.yml]: v0.5.0-rancher1

**Comment by janeczku** [comment]
LGTM @vincent99 @alena1108

**Pull request merged**